cmd/artifact_install.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2022 The Falco Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/spf13/cobra"
+
+	"github.com/falcosecurity/falcoctl/cmd/internal/utils"
+	"github.com/falcosecurity/falcoctl/pkg/index"
+	"github.com/falcosecurity/falcoctl/pkg/oci"
+	"github.com/falcosecurity/falcoctl/pkg/oci/authn"
+	ocipuller "github.com/falcosecurity/falcoctl/pkg/oci/puller"
+	"github.com/falcosecurity/falcoctl/pkg/options"
+)
+
+const (
+	defaultPluginsDir    = "/usr/share/falco/plugins"
+	defaultRulesfilesDir = "/etc/falco"
+)
+
+type artifactInstallOptions struct {
+	*options.CommonOptions
+	credentialStore *authn.Store
+	rulesfilesDir   string
+	pluginsDir      string
+}
+
+// NewArtifactInstallCmd returns the artifact search command.
+func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cobra.Command {
+	o := artifactInstallOptions{
+		CommonOptions: opt,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "install [ref1 [ref2 ...]] [flags]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Install a list of artifacts",
+		Long:                  "Install a list of artifacts",
+		Args:                  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			o.Printer.CheckErr(o.RunArtifactInstall(ctx, args))
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.rulesfilesDir, "rulesfiles-dir", "", defaultRulesfilesDir,
```

Comment:
If we use only one flag, what happens if the same install command is used to install both a plugin and a rule at the same time? I think it is better to keep these flags separated also because when we will be able to handle dependencies (subsequent PRs), we will have ways to say where artifacts should be moved independently. But I am open to modify this if there are other reasons to do that!